### PR TITLE
NIFI-2632: Added fragment attributes to SplitJson and SplitXml

### PR DESCRIPTION
**Pull request by mattyb149**

### CODE REVIEW

**Comment by YolandaMDavis** [comment]
@mattyb149 I can review

File: nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/SplitJson.java [hunk 1]
```diff
@@ -61,6 +63,16 @@
         + "Each generated FlowFile is comprised of an element of the specified array and transferred to relationship 'split,' "
         + "with the original file transferred to the 'original' relationship. If the specified JsonPath is not found or "
         + "does not evaluate to an array element, the original file is routed to 'failure' and no files are generated.")
+@WritesAttributes({
+        @WritesAttribute(attribute = "fragment.identifier",
```
Comment (YolandaMDavis) [inline nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/SplitJson.java]: Suggest a different name to indicate this is more of a fragment from a the same parent (e.g. fragment.parent.identifier or fragment.group.identifier)?  To me using fragment alone implies a unique id for that split instance (even though I know the attribute description explains more)

Comment (mattyb149) [inline nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/SplitJson.java]: I don't like the name either, but wanted to keep it consistent with the other split processors

**Comment by YolandaMDavis** [comment]
@mattyb149 Understood on the above comment.  Just to note it (for maybe future refactor) it may be cool to have some abstract SplitProcessor that has attributes, relationships and some of the split/original flow file handling that seems to be similar across many of the split processors.

This looks good tested with flows using json and xml. Splits were executed as expected.  Extracted attributes in both cases and confirmed new attributes for flow files transferred to split queue (original files did not contain attributes).  Attributes were also available for expression languages.

+1

I'll merge this in shortly

**Pull request closed without merging**